transports/onion/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+#![doc(html_logo_url = "https://libp2p.io/img/logo_small.png")]
+#![doc(html_favicon_url = "https://libp2p.io/img/favicon.png")]
+#![cfg_attr(docsrs, feature(doc_cfg))]
+//! Tor based transport for libp2p. Connect through the Tor network to TCP listeners.
+//!
+//! Main entrypoint of the crate: [OnionTransport]
+//!
+//! ## Example
+//! ```no_run
+//! # use async_std_crate as async_std;
+//! # use libp2p_core::Transport;
+//! # async fn test_func() -> Result<(), Box<dyn std::error::Error>> {
+//! let address = "/dns/www.torproject.org/tcp/1000".parse()?;
+//! let mut transport = libp2p_onion::AsyncStdNativeTlsOnionTransport::bootstrapped().await?;
+//! // we have achieved tor connection
+//! let _conn = transport.dial(address)?.await?;
+//! # Ok(())
+//! # }
+//! # async_std::task::block_on(async { test_func().await.unwrap() });
```

Comment:
Would you mind fixing the indentation here? As far as I know it will propagate to the rendered HTML.

Reply:
<img width="979" alt="image" src="https://user-images.githubusercontent.com/55623006/192736528-c92f412c-fd1f-4e5d-84ea-3faed1030fe1.png">

I don't see any indentation issues, if you see some, i will correct them.